ldb/views.py

```diff
@@ -39,11 +37,29 @@ class ResultsView(TemplateView):
 
     def get_results(self):
         q = self.request.GET.get("q")
+        print(q)
         if q is not None:
-            return (
-                SearchQuerySet()
-                .models(Person, Organization)
-                .filter(SQ(text__icontains=q))
+            q = q.strip().split()
+
+            person_filters = Q()
+            organization_filters = Q()
+            for word in q:
+                person_filters &= (
+                    Q(firstname__icontains=word)
+                    | Q(preposition__icontains=word)
+                    | Q(surname__icontains=word)
+                    | Q(ldap_username__icontains=word)
+                    | Q(street_name__icontains=word)
+                )
+                organization_filters &= Q(name__icontains=word)
+
+            # Return all persons
+            return list(
+                chain(
+                    # Filter person on firstname, surname, email and address
+                    Person.objects.filter(person_filters),
+                    Organization.objects.filter(organization_filters),
+                )
```

Comment:
Hmmm... by converting this to a list Django will return all results from the database. Later we strip this by only returning the first 10 elements.

limiting both queries to 10 (by adding `[:10]` after the `.filter()`) could be a quickfix. However, then the `count` won't be correct anymore. 

Also: I see the count is achieved by using `len()` it is beter to `.count()` on the queryset. (maybe the counts should also be added to this method and returned separately?)